### PR TITLE
 Closes #1150: Fix issues with Single Card paragraph migration.

### LINKED ISSUE #1150: AZ Paragraph Single Card Migration plugin empty link issue
## Problem/Motivation
Migration failed on the Single Card migrate.

### Describe the bug
I tracked down a few of the errors, and found that each of the cards had no links, where as other cards on the same page seemed to pass.
Error note:
```
Array index missing, extraction failed for 'array()'. Consider adding a `default` key to the configuration.
```

### To Reproduce
1-Make a new single card, and don't give it a link. It will fail on migrate.
2-Made a test card first with a link, and did not get an error. Proves link may be causing it.

## Proposed resolution
Check out the error note description and have case for no links

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Fixes issues that were preventing the Single Card paragraph migration from working correctly.

### _Removed (will be addressed in #1201)_:
Note: this adds a migrated path lookup step to the link field on Single cards which means that the Single Card migration needs to be ran a second time (in update mode) after the node and term migrations have been run to update the migration lookup stubs:

```
$ drush mim --group az_migration
...
(lots of progress messages)
...

$ drush mim az_paragraph_single_card --update
  [notice] Processed 17 items (0 created, 17 updated, 0 failed, 0 ignored) - done with 'az_paragraph_single_card'
```

The term and node migrations used in the migrated path lookup step are added as optional dependencies because the single card migration is a dependency for the flexible page migration and so it can't be added as a required dependency to this migration. 

## Related Issue
#1150 

## How Has This Been Tested?
Locally with lando using a dev copy of the quickstart1.arizona.edu site as the source.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml [hunk 1]
```diff
@@ -15,50 +15,95 @@ destination:
...
+        - az_flexible_page_categories
+        - az_news_tags
+        - az_person_categories_secondary
+        - az_person_categories
+      node_migration:
+        - az_node_carousel
+        - az_node_event
+        - az_node_flexible_page
+        - az_node_news
+        - az_node_person
```
Comment (trackleft) [inline modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml:76]: Ha, links strike again

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml [hunk 2]
```diff
@@ -15,50 +15,95 @@ destination:
...
+
   behavior_settings:
     plugin: paragraphs_behavior_settings
     card_width: 'col-md-12 col-lg-12'
     card_style: 'card'
     card_width_sm: 'col-sm-12'
     card_width_xs: 'col-12'
 
+migration_dependencies:
+  optional:
```
Comment (trackleft) [inline modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_single_card.yml:95]: Oh, cool

**Comment by trackleft** [comment]
Circular dependency aside, I think this is what we want.

**Review by trackleft — APPROVED** [review]

**Comment by joeparsons** [comment]
Removed migrated path lookup step for now (will be addressed in #1201).

**Review by tadean — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**